test/powershell/Modules/Microsoft.PowerShell.Management/Get-PSProvider.Tests.ps1

```diff
@@ -24,4 +24,56 @@ Describe "Get-PSProvider" -Tags "CI" {
 
 	{ $actual | Format-List } | Should -Not -Throw
     }
+
+    Context 'PathSeparators' {
+        BeforeAll {
+            $windowsTestCases = @(
+                @{Provider = 'FileSystem'; Value = @("\", "/")}
+                @{Provider = 'Variable'; Value = @("\", "/")}
+                @{Provider = 'Function'; Value = @("\", "/")}
+                @{Provider = 'Registry'; Value = @("\")}
+            )
+
+            $nonWindowsTestCases = @(
+                @{Provider = 'FileSystem'; Value = @("/", "\")}
+                @{Provider = 'Variable'; Value = @("/", "\")}
+                @{Provider = 'Function'; Value = @("/", "\")}
+            )
+
+            $testCases = if ($IsWindows) {
+                            $windowsTestCases
+                        }
+                        else {
+                            $nonWindowsTestCases
+                        }
+        }
+
+        It '<Provider> provider has PathSeparator property' -TestCases $testCases {
+            param ($Provider, $Value)
+
+            (Get-PSProvider $Provider).PathSeparator | Should -Be $Value
+        }
+
+        It 'cannot modify PathSeparator collection in <Provider> provider' -TestCases $testCases {
+            param ($Provider, $Value)
+
+            $separator = (Get-PSProvider $Provider).PathSeparator
+
+            { $separator[0] = "w" } | Should -Throw
+        }
+
+        It 'copying and modifying values does not affect PathSeparator property in <Provider> provider' -TestCases $testCases {
+            param ($Provider, $Value)
```

Comment:
This test case doesn't seem to be very interesting. Changing the copy of a collection doesn't affect the original collection.

Reply:
Now that we're using a `ReadOnlyCollection` probably not, I put it in place to provide a case that if we were to use an array instead, it would be possible to modify the content of the PathSeparator in the same powershell session

Reply:
Fair point.